### PR TITLE
make spec great again

### PR DESCRIPTION
**Pull request by AnEmortalKid**
rewrote the spec to have more examples and more explanation.

I recommend viewing the produced markdown (and not the diff ) since the diff is hard to follow.

### CODE REVIEW

**Comment by AnEmortalKid** [comment]
adding @alistairyves  and @Bartvds in case.

**Comment by Bartvds** [comment]
@AnEmortalKid Please go ahead with what you think is best to revive this project 👍 

If you need anything specific from me please ping and mail.
